pkg/operators/v1alpha1/catalogsource_types.go

```diff
@@ -113,10 +120,45 @@ type GrpcPodConfig struct {
 	// +optional
 	PriorityClassName *string `json:"priorityClassName,omitempty"`
 
-	// RunAsRoot allows admins to indicate that they wish to run the container inside the CatalogSource pod in a privileged
-	// mode as root.  This should only be enabled when running older catalog images which could not be run as non-root.
+	// SecurityContextConfig can be one of "legacy" or "restricted". The grpc pod created is either injected with
+	// the right pod.spec.securityContext and pod.spec.container[*].securityContext values to allow the pod to run in
+	// Pod Security Admission(PSA) "restricted" mode, or does not set these values at all, in which case the pod
+	// can only be run in PSA "privileged" namespaces. By default, SecurityContextConfig is set to "restricted". If the value
+	// is unspecified, the default value of restricted is used.  Specifying any other value will result in a validation error.
+	// When using older catalog images, which could not be run in restricted mode, the SecurityContextConfig should be set to
+	// "legacy". More information about PSA can be found here: https://kubernetes.io/docs/concepts/security/pod-security-admission/
 	// +optional
-	RunAsRoot bool `json:"runAsRoot,omitempty"`
+	SecurityContextConfig SecurityConfig `json:"securityContextConfig,omitempty"`
+}
+
+// UnmarshalJSON for GrpcPodConfig is a custom unmarshaler that
+// sets the default value of `restricted` for SecurityContextConfig
+// if a value isn't set explicitly, and throws an error if an incorrect
+// value (a value other than `restricted` or `legacy` ) is set for
+// SecurityContextConfig
+func (g *GrpcPodConfig) UnmarshalJSON(data []byte) (err error) {
```

Comment:
Haven't we run into major issues with relying on `UnmarshalJSON`-based defaulting and validating? IIRC, we had a problem with the catalog source polling interval duration value where an invalid catsrc existing in etcd caused the entire informer to fail, making it impossible to reconcile all catalog sources.

Can we instead use CRD schema defaulting and validating?

Reply:
@joelanford done.